### DISCUSSION TITLE
How to replace a broken "patch" command on RH6

### DISCUSSION THREAD

**Discussion by nkgh77** [spack/spack#28963] [Q&A] [answered]
I have encountered a problem that is caused by a broken "patch" command, which is provided by RH6.
Looking at a comment, the patch command is a spack-wide issue, not just a package.
Is there a good way to use a good patch command in SPACK?

https://github.com/spack/spack/issues/28776#issuecomment-1040353063

**Comment by haampie** [comment, marked as answer]
Maybe we should bootstrap patch @alalazo?

At the very least we can check if the user version of `patch` is available and free of bugs (i.e. >= 2.7).

But @nkgh77, for now you can build patch with spack: `spack install patch@2.7:` and then put the `/path/to/spacks/version/of/patch/bin` in your PATH.

**Reply by alalazo** [reply]
> Maybe we should bootstrap patch @alalazo?

In principle we should bootstrap all of [these things](https://spack.readthedocs.io/en/latest/getting_started.html#system-prerequisites) which are now listed as system requirements. We can for sure add a minimum version to the docs, if we know some versions are broken. A further step might be adding bootstrap from sources capabilities to Spack.
